src/Validation/Constraints/HasMaxLength.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/* Copyright (c) 2017 Jesús López <lopez@leifos.com> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Validation\Constraints;
+use ILIAS\Validation\Constraint;
+use ILIAS\Data;
+
+class HasMaxLength extends Custom implements Constraint {
+	/**
+	 * @var int
+	 */
+	protected $max_length;
+
+	//TODO-> use lang vars.
+	public function __construct($max_length, Data\Factory $data_factory) {
+		assert('is_int($max_length)');
+		$this->max_length = $max_length;
+		parent::__construct( function ($value) {
+			return strlen($value) <= $this->max_length;
+		},
+			function () {
+				return "The entered text has a length more than '{$this->max_length}'.";
```

Comment:
Hi @Amstutz, @klees  All the UI Constraints are using hard coded messages(english), no one is using lang vars. Is multi-language available somehow? is this a bug or it's something that are you working on?

Reply:
This is a issue I'm working on. I already have some code ready, but need [this](https://github.com/ILIAS-eLearning/ILIAS/pull/1054) to make it work. So no need for you to i18n that by yourself.